symbolic-symcache/src/writer.rs

```diff
@@ -199,19 +191,16 @@ impl SymCacheConverter {
 
             let string_bytes = &mut self.string_bytes;
             let strings = &mut self.strings;
-            let path_name_offset = Self::insert_string(string_bytes, strings, &location.file.name);
-            let directory_offset = location
-                .file
-                .directory
-                .map_or(u32::MAX, |d| Self::insert_string(string_bytes, strings, &d));
-            let comp_dir_offset = location.file.comp_dir.map_or(u32::MAX, |cd| {
-                Self::insert_string(string_bytes, strings, &cd)
+            let name_offset = Self::insert_string(string_bytes, strings, &location.file.name);
+            let directory_offset = location.file.directory.map_or(u32::MAX, |d| {
+                let comp_dir = std::str::from_utf8(function.compilation_dir).unwrap_or_default();
+                let joined = join_path(comp_dir, &d);
+                let cleaned = clean_path(&joined);
+                Self::insert_string(string_bytes, strings, &clean_path(&cleaned))
```

Comment:
I was proposing to do the 3-way join/clean, and then split that result into dirname/basename; and persist that into the symcache.

It would be interesting to have some specific examples of what kind of inputs we get from real-world C projects.

Also, we should at least think about how to make this fast. Doing a 3-way merge and split into dirname/basename could potentially be expensive; as we run this piece of code for each source location. Some smart caching could speed this up potentially?